server/posts/views.py

```diff
@@ -25,14 +25,23 @@ class UpdatePostView(generics.RetrieveUpdateDestroyAPIView): #mixins.DestroyMode
     # returns 404 otherwise
     def get_post(self):
         pk = self.kwargs.get('')
-        request_author_id = self.kwargs['author_id']
+        request_author_id = uuid.UUID(self.kwargs['author_id'])
         request_post_id = self.kwargs['pk']
+        sharer_id = self.request.user.id
+        try:
+            a_post = Post.objects.get(pk=uuid.UUID(request_post_id))
+        except Post.DoesNotExist:
+            try:
+                sharer_items = Inbox.objects.get(author=sharer_id).items
+            except Inbox.DoesNotExist:
+                return Response({'error': 'Inbox not found!'},
```

Comment:
Have you tested this response? How does it work? Will `def retrieve()` understand how to place a Response inside a Response?